operator/api/v1alpha4/coreprovider_types.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha4
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CoreProviderSpec defines the desired state of CoreProvider
+type CoreProviderSpec struct {
+	ProviderSpec `json:",inline"`
+}
+
+// CoreProviderStatus defines the observed state of CoreProvider
+type CoreProviderStatus struct {
+	ProviderStatus `json:",inline"`
+}
+
+// +kubebuilder:object:root=true
+
+// CoreProvider is the Schema for the coreproviders API
+type CoreProvider struct {
```

Comment:
Hmm...usually when I hear CAPI Providers (_in plural_), I think of the core provider and the kubeadm bootstrap/control plane providers as well. Maybe because I know they are all within the same repo. 😄 

If other maintainers/reviewers think CAPIProvider makes more sense, I'm fine with updating it. Personally, I think `CoreProvider` makes sense 🙂 

We may also have to update our documentation like https://cluster-api.sigs.k8s.io/clusterctl/commands/init.html?highlight=core#automatically-installed-providers where we refer to core provider.

Reply:
Instead of having a type explicitly for "CoreProvider", where we expect there to only be a singleton instance, would it make more sense to have a "ManagementCluster" type, that embeds the config we expect for the core provider, but could also provide summarized status for all of the other types of providers that make up the Management Cluster?

Reply:
The idea of a ManagementCluster object is intriguing, but I'm ok with CoreProvider as defined in the proposal for the first iteration